### PR TITLE
Implement advanced copy rules

### PR DESCRIPTION
**Pull request by innovimax-jim**

### CODE REVIEW

**Comment by innovimax-jim** [comment]
Refers to #130

**Comment by innovimax-jim** [comment]
Implemented rule -file #index to output the nth file

File: src/test/resources/innovimax/mixthem/test021_params-file.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+3th 3
```
Comment (innovimax) [inline src/test/resources/innovimax/mixthem/test021_params-file.txt]: why 3*th* ??? it should be just "3"

File: src/main/java/innovimax/mixthem/MixThem.java [hunk 2]
```diff
@@ -105,6 +105,10 @@ public void process(FileMode fileMode, Rule rule, Map<RuleParam, ParamValue> par
                     final IOperation copyFile2Op = CopyFactory.newInstance(fileMode, CopyMode.SECOND, params);
                     copyFile2Op.processFiles(this.inputs, this.output);          
                     break;
+                case FILE_N:
```
[the remark this answers is not in the document]
Comment (innovimax-jim) [reply src/main/java/innovimax/mixthem/MixThem.java]: ok

File: src/test/resources/innovimax/mixthem/test021_params-file.txt (hunk 1 above)
Comment (innovimax-jim) [reply src/test/resources/innovimax/mixthem/test021_params-file.txt]: It is just a free suffix to distinguish test rule runs and that is appended to the name of result file used for the test

**Comment by innovimax-jim** [comment]
Implemented + [#list] rule to output a selection of files

File: src/main/java/innovimax/mixthem/operation/AbstractCopyOperation.java [hunk 3]
```diff
@@ -42,15 +44,25 @@ public void processFiles(final List<InputResource> inputs, final OutputStream ou
 			case SECOND:
 				process(inputs.get(1), output);
 				break;
+			case UMPTEENTH:
```
[the remark this answers is not in the document]
Comment (innovimax-jim) [reply src/main/java/innovimax/mixthem/operation/AbstractCopyOperation.java]: This is the copy mode that indicates to output the umpteenth file ; the index of this file is given by the FILE_INDEX rule parameter.

File: src/main/java/innovimax/mixthem/arguments/Rule.java [hunk 4]
```diff
@@ -10,12 +10,14 @@
 * @version 1.0
 */
 public enum Rule { 
-    FILE_1("1", "1", "will output file1", true, 
+    FILE_1("1", "1", "will output first file", true, 
                 EnumSet.noneOf(RuleParam.class), EnumSet.of(FileMode.CHAR, FileMode.BYTE)),
```
Comment (innovimax) [inline src/main/java/innovimax/mixthem/arguments/Rule.java]: Remove FILE_1 and FILE_2 since they are sepcial case of FILE_K

Comment (innovimax-jim) [reply src/main/java/innovimax/mixthem/arguments/Rule.java]: ok

**Comment by innovimax-jim** [comment]
Deleted rules -1 & -2 (replaced by -file)

**Pull request merged**